client/src/components/payment/payment-summary.vue

```diff
@@ -1,7 +1,7 @@
 <template>
   <v-expand-transition>
-    <!-- do not display until payments are fetched -->
-    <section class="payment-summary" v-if="summary">
+    <!-- check for valid receipt, as the payment may exist but not be completed -->
+    <section class="payment-summary" v-if="summary.receipt">
```

Comment:
Quick question, just to be thorough, should this be v-if="summary && summary.receipt" ?

Reply:
Good question!

This component is in a v-for loop that iterates over summaries (and summary, a prop, is each item). So in theory this component is not rendered if the loop is empty, and summary is always defined. I tested a few scenarios locally to verify that this component doesn't have an undefined/null summary object.

Note that some components in Namerequest are pre-mounted and are then conditionally displayed. In that case, the component may exist without valid props. But not in this case (I think).

If you insist, I can add the original check back in. But I'd be more curious to see if it ever fails and then handle THAT situation :)

Reply:
got it & do not insist ;)